Framework/Core/src/SendingPolicy.cxx

```diff
@@ -28,8 +28,22 @@ std::vector<SendingPolicy> SendingPolicy::createDefaultPolicies()
             .name = "dispatcher",
             .matcher = [](DeviceSpec const& spec, ConfigContext const&) { return spec.name == "Dispatcher" || DeviceSpecHelpers::hasLabel(spec, "Dispatcher"); },
             .send = [](FairMQDeviceProxy& proxy, fair::mq::Parts& parts, ChannelIndex channelIndex, ServiceRegistryRef registry) {
-              auto *channel = proxy.getOutputChannel(channelIndex);
-              channel->Send(parts, -1); }},
+              OutputChannelInfo const& info = proxy.getOutputChannelInfo(channelIndex);
+              OutputChannelState& state = proxy.getOutputChannelState(channelIndex);
+              // Default timeout is 10ms.
+              // We count the number of consecutively dropped messages.
+              // If we have more than 10, we switch to a completely
+              // non-blocking approach.
+              int64_t timeout = 10;
+              if (state.droppedMessages > 10) {
```

Comment:
I like the proposition by David, but perhaps a warning could be added when we switch to fully non-blocking?
```suggestion
              int64_t timeout = 10;
              if (state.droppedMessages == 10 + 1) {
                LOGP(warning) << "Failed to send 10 messages with 10ms timeout, switching to completely non-blocking mode"; 
              } 
              if (state.droppedMessages > 10) {
```